crates/ty_project/src/metadata/options.rs

```diff
@@ -226,17 +258,26 @@ impl Options {
     }
 }
 
-#[derive(Debug, Default, Clone, Eq, PartialEq, Combine, Serialize, Deserialize)]
+#[derive(
+    Debug, Default, Clone, Eq, PartialEq, Combine, Serialize, Deserialize, OptionsMetadata,
+)]
 #[serde(rename_all = "kebab-case", deny_unknown_fields)]
 #[cfg_attr(feature = "schemars", derive(schemars::JsonSchema))]
 pub struct EnvironmentOptions {
     /// Specifies the version of Python that will be used to analyze the source code.
     /// The version should be specified as a string in the format `M.m` where `M` is the major version
-    /// and `m` is the minor (e.g. "3.0" or "3.6").
+    /// and `m` is the minor (e.g. `"3.0"` or `"3.6"`).
     /// If a version is provided, ty will generate errors if the source code makes use of language features
     /// that are not supported in that version.
     /// It will also tailor its use of type stub files, which conditionalizes type definitions based on the version.
```

Comment:
Not changed in this PR, but our understanding of `sys.version_info` branches is not limited to type stub files, it works just as well in code files. Suggested update:
```suggestion
    /// It will also understand conditionals based on comparisons with `sys.version_info`, such
    /// as are commonly found in typeshed to reflect the differing contents of the standard
    /// library across Python versions.
```
Same thing would apply to the similar comment below on platform (which I can't inline-comment on, since it's out of the modified range of this PR.)